editor/src/components/editor/store/dispatch.tsx

```diff
@@ -94,11 +94,16 @@ import { isFeatureEnabled } from '../../../utils/feature-switches'
 import { isJsOrTsFile, isCssFile } from '../../../core/shared/file-utils'
 import { applyStatePatches } from '../../canvas/commands/commands'
 
-export interface DispatchResult extends EditorStore {
+interface DispatchResultFields {
   nothingChanged: boolean
   entireUpdateFinished: Promise<any>
 }
 
+type EditorStoreUnpatched = Omit<EditorStoreFull, 'patchedEditor'>
+
+export interface InnerDispatchResult extends EditorStoreUnpatched, DispatchResultFields {}
```

Comment:
this is probably my own personal preference but I find this form to be vastly more readable:

```suggestion
export type InnerDispatchResult = EditorStoreUnpatched & DispatchResultFields 
```

Reply:
Done